todo_app/app.py

```diff
@@ -1,35 +1,94 @@
-from flask import Flask, redirect, render_template, request
+from cmath import log
+from functools import wraps
+import json
+import os
+import requests
+from flask import Flask, abort, redirect, render_template, request
 from dotenv import load_dotenv
+from flask_login import LoginManager, current_user, login_required, login_user
+from oauthlib.oauth2 import WebApplicationClient
 from todo_app.data.item import ItemStatus
 from todo_app.data.mongo_items import get_item, get_items, add_item, delete_item, save_item
 from todo_app.data.itemsViewModel import ItemsViewModel
+from todo_app.data.user import User, Role, user_id_to_role
 load_dotenv()
 
+def authorised_role(role):
+    def decorator(f):
+        @wraps(f)
+        def decorated_function(*args, **kwargs):
+            user_id = getattr(current_user, "id", None)
+            user_role = user_id_to_role.get(str(user_id), Role.READER)
+
+            login_disabled = os.getenv('LOGIN_DISABLED') == 'True'
+            if (not login_disabled and role == Role.WRITER and user_role != Role.WRITER):
```

Comment:
Could just use `current_user.role` instead of creating this `user_role` variable